Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -1332,22 +1332,27 @@ namespace OpenInfraPlatform {
 									loops.push_back(loop);
 								}
 								
-								std::vector<carve::geom::vector<3>> firstPointloop = loops[0];
-								std::vector<carve::geom::vector<3>> secondPointLoop = loops[1];
-
-								carve::geom::vector<3> firstPoint = firstPointloop[0];
-								carve::geom::vector<3> secondPoint = firstPointloop[1];
-								carve::geom::vector<3> thirdPoint = firstPointloop[2];
-
-								carve::geom::vector<3> firstToSecond = secondPoint - firstPoint;
-								carve::geom::vector<3> firstToThird = thirdPoint - firstPoint;
-
-								carve::geom::vector<3> normalOfPlane = carve::geom::cross(firstToSecond, firstToThird);
-								normalOfPlane.normalized();
-								
+								std::vector<std::vector<carve::geom::vector<2>>> loops2D;
+								carve::geom::vector<3> normalOfPlane = GeomUtils::computePolygonNormal(loops[0]);
 
+								for (int i = 0; i < loops.size(); i++) 
+								{
+									std::vector<carve::geom::vector<3>> loop = loops[i];

```

Comment:
similarly:

`for( const auto loop : loops )`

Reply:
Resolved:
7a17b33